codegen/datamodelgen.py

```diff
@@ -103,7 +103,7 @@ def __init__(
             / f"datamodel_{version}"
         ).resolve()
         datamodel_dir.mkdir(exist_ok=True)
-        self.filepath = (datamodel_dir / f"{rules_save_name}.py").resolve()
+        self.file_name = (datamodel_dir / f"{rules_save_name}.py").resolve()
```

Comment:
These is actually a file path. Also, the version from `get_version_for_filepath` is appended to both file names and folder names. So, better to change these to `file_path`.